### PR TITLE
fix(scripts): optimize PSScriptAnalyzer linting performance in WSL2

### LINKED ISSUE #674: PSScriptAnalyzer linting takes ~35 minutes in WSL2 due to recursive file discovery and PATH scanning
## Description

`npm run lint:ps` takes approximately 35 minutes to complete in WSL2 environments, making local development iteration impractical. The expected runtime for 52 PowerShell files is under 1 minute.

### 1. Recursive file discovery via `Get-ChildItem`

`Get-FilesRecursive` in `LintingHelpers.psm1` uses `Get-ChildItem -Recurse` to find PowerShell files. This traverses the entire filesystem tree (12,040+ items including `node_modules/`, `.git/`, and other irrelevant directories) to locate 52 PowerShell files. The gitignore filtering happens *after* enumeration, so the full traversal cost is always paid.

### 2. WSL2 PATH containing Windows mount points

WSL2 environments typically have 30+ `/mnt/*` entries in `$env:PATH` (Windows system directories shared via Plan 9/9P protocol over Hyper-V AF_VSOCK sockets). PSScriptAnalyzer/.NET resolves commands by scanning every PATH directory for each file analyzed. Each `/mnt/*` lookup triggers a cross-filesystem `openat` syscall through the 9P protocol, adding ~0.5-1 second per file analyzed.

Combined, these cause ~5,600 slow cross-filesystem syscalls per linting run, confirmed via `strace`.

## Fix

PR #667 addresses both root causes:

- **`Get-FilesRecursive`**: Uses `git ls-files --cached --others --exclude-standard` (scoped to the `$Path` parameter) with fallback to `Get-ChildItem -Recurse` for non-git contexts
- **`Invoke-PSScriptAnalyzer.ps1`**: Strips `/mnt/*` entries from PATH before running analysis (guarded by the direct-invocation check so it does not affect dot-sourced test imports)

## Validation

After fix: 52 files analyzed, 0 issues, ~40 seconds (down from ~35 minutes).

### PR DESCRIPTION
**Pull request by WilliamBerryiii**
## Description

Resolved two independent root causes of extreme PSScriptAnalyzer slowness in WSL2 environments. File discovery used `Get-ChildItem -Recurse`, traversing 12,000+ filesystem items to find ~52 PowerShell files. PATH scanning by PSScriptAnalyzer/.NET resolved commands against ~36 inherited `/mnt/*` Windows mount entries, each incurring slow 9P cross-filesystem lookups via AF_VSOCK. Together these caused `npm run lint:ps` to take upwards of 35 minutes. After the fix, the same run completes in ~40 seconds.

- perf(scripts): replaced `Get-ChildItem -Recurse` with `git ls-files --cached --others --exclude-standard` in `Get-FilesRecursive`, scoped to the requested `$Path` parameter and filtered by `$Include` patterns
- perf(scripts): added PATH sanitization in `Invoke-PSScriptAnalyzer.ps1` main execution block to strip `/mnt/*` entries before analysis, guarded so it only runs on direct invocation
- fix(scripts): scoped `git ls-files` results to the requested `$Path` parameter and restored gitignore filtering in the `Get-ChildItem` fallback path
- refactor(scripts): removed dead `$gitignorePath` variable from `Invoke-PSScriptAnalyzerCore`
- test(scripts): added 7 tests for `Get-FilesRecursive` covering git ls-files scoping, subdirectory filtering, fallback behavior, and backward compatibility
- test(scripts): added 4 tests for PATH sanitization logic covering `/mnt/*` stripping, preservation of native paths, and edge cases
- docs: updated `Get-FilesRecursive` descriptions in `scripts/linting/README.md` and `docs/architecture/testing.md` to reflect `git ls-files` primary path with `Get-ChildItem` fallback

## Related Issue(s)

Fixes #674

## Type of Change

Select all that apply:

**Code &amp; Documentation:**

- [x] Bug fix (non-breaking change fixing an issue)
- [ ] New feature (non-breaking change adding functionality)
- [ ] Breaking change (fix or feature causing existing functionality to change)
- [x] Documentation update

**Infrastructure &amp; Configuration:**

- [ ] GitHub Actions workflow
- [ ] Linting configuration (markdown, PowerShell, etc.)
- [ ] Security configuration
- [ ] DevContainer configuration
- [ ] Dependency update

**AI Artifacts:**

- [ ] Reviewed contribution with `prompt-builder` agent and addressed all feedback
- [ ] Copilot instructions (`.github/instructions/*.instructions.md`)
- [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
- [ ] Copilot agent (`.github/agents/*.agent.md`)
- [ ] Copilot skill (`.github/skills/*/SKILL.md`)

&gt; **Note for AI Artifact Contributors**:
&gt;
&gt; - **Agents**: Research, indexing/referencing other project (using standard VS Code GitHub Copilot/MCP tools), planning, and general implementation agents likely already exist. Review `.github/agents/` before creating new ones.
&gt; - **Skills**: Must include both bash and PowerShell scripts. See [Skills](../docs/contributing/skills.md).
&gt; - **Model Versions**: Only contributions targeting the **latest Anthropic and OpenAI models** will be accepted. Older model versions (e.g., GPT-3.5, Claude 3) will be rejected.
&gt; - See [Agents Not Accepted](../docs/contributing/custom-agents.md#agents-not-accepted) and [Model Version Requirements](../docs/contributing/ai-artifacts-common.md#model-version-requirements).

**Other:**

- [x] Script/automation (`.ps1`, `.sh`, `.py`)
- [ ] Other (please describe):

## Testing

- Ran `npm run lint:ps` end-to-end: 52 files analyzed, 0 issues, ~40 seconds (down from ~35 minutes)
- Ran `npm run test:ps`: all 45 linting tests pass (25 LintingHelpers + 20 Invoke-PSScriptAnalyzer)
- Verified `git ls-files` scoping returns only files under the requested `$Path`
- Verified `Get-ChildItem` fallback activates when `git rev-parse` fails
- Verified PATH sanitization strips `/mnt/*` entries while preserving native Linux paths

### Required Checks

- [x] Documentation is updated (if applicable)
- [x] Files follow existing naming conventions
- [x] Changes are backwards compatible (if applicable)
- [x] Tests added for new functionality (if applicable)

### AI Artifact Contributions

- [ ] Used `/prompt-analyze` to review contribution
- [ ] Addressed all feedback from `prompt-builder` review
- [ ] Verified contribution follows common standards and type-specific requirements

### Required Automated Checks

The following validation commands must pass before merging:

- [ ] Markdown linting: `npm run lint:md`
- [ ] Spell checking: `npm run spell-check`
- [ ] Frontmatter validation: `npm run lint:frontmatter`
- [ ] Skill structure validation: `npm run validate:skills`
- [ ] Link validation: `npm run lint:md-links`
- [x] PowerShell analysis: `npm run lint:ps`

## Security Considerations

- [x] This PR does not contain any sensitive or NDA information
- [ ] Any new dependencies have been reviewed for security issues
- [x] Security-related scripts follow the principle of least privilege

## Additional Notes

The PATH sanitization only affects the `$env:PATH` within the PowerShell process when the script is invoked directly (guarded by `$MyInvocation.InvocationName -ne '.'`). It does not modify the system PATH or affect other processes.

The `Get-FilesRecursive` function maintains backward compatibility: the `$GitIgnorePath` parameter is still accepted but only used in the `Get-ChildItem` fallback path. The `git ls-files --exclude-standard` flag provides equivalent gitignore filtering natively.

⚡ - Generated by Copilot

### CODE REVIEW

File: scripts/linting/Modules/LintingHelpers.psm1 [hunk 1]
```diff
@@ -116,25 +116,28 @@ function Get-FilesRecursive {
...
-            foreach ($pattern in $gitignorePatterns) {
-                if ($file.FullName -like $pattern) {
-                    $excluded = $true
-                    break
-                }
-            }
-            
-            -not $excluded
+    # git ls-files handles gitignore natively; parameter kept for callers
+    $null = $GitIgnorePath
```
Comment (katriendg) [inline scripts/linting/Modules/LintingHelpers.psm1]: Do we need to update docs about this change?

Comment (WilliamBerryiii) [reply scripts/linting/Modules/LintingHelpers.psm1]: This is resolved
